pyccl/halos/hmfunc.py

```diff
@@ -0,0 +1,673 @@
+
+
+class MassFuncTinker10(MassFunc):
+    """ Implements mass function described in arXiv:1001.3162.
+
+    Args:
+        cosmo (:obj:`Cosmology`): A Cosmology object.
+        mass_def (:obj:`MassDef`): a mass definition object.
+            this parametrization accepts SO masses with
+            200 < Delta < 3200 with respect to the matter density.
+            If `None`, Delta = 200 (matter) will be used.
+    """
+    name = 'Tinker10'
+
+    def __init__(self, cosmo, mass_def=None):
+        super(MassFuncTinker10, self).__init__(cosmo,
+                                               mass_def)
+
+    def _default_mdef(self):
+        self.mdef = MassDef200m()
+
+    def _setup(self, cosmo):
+        from scipy.interpolate import interp1d
+
+        delta = np.array([200.0, 300.0, 400.0, 600.0, 800.0,
+                          1200.0, 1600.0, 2400.0, 3200.0])
+        delta = np.array([200.0, 300.0, 400.0, 600.0, 800.0,
+                          1200.0, 1600.0, 2400.0, 3200.0])
+        alpha = np.array([0.368, 0.363, 0.385, 0.389, 0.393,
+                          0.365, 0.379, 0.355, 0.327])
+        beta = np.array([0.589, 0.585, 0.544, 0.543, 0.564,
+                         0.623, 0.637, 0.673, 0.702])
+        gamma = np.array([0.864, 0.922, 0.987, 1.09, 1.20,
+                          1.34, 1.50, 1.68, 1.81])
+        phi = np.array([-0.729, -0.789, -0.910, -1.05, -1.20,
+                        -1.26, -1.45, -1.50, -1.49])
+        eta = np.array([-0.243, -0.261, -0.261, -0.273, -0.278,
+                        -0.301, -0.301, -0.319, -0.336])
+
+        ldelta = np.log10(delta)
+        ld = np.log10(self.mdef.Delta)
+        self.pA0 = interp1d(ldelta, alpha)(ld)
+        self.pa0 = interp1d(ldelta, eta)(ld)
+        self.pb0 = interp1d(ldelta, beta)(ld)
+        self.pc0 = interp1d(ldelta, gamma)(ld)
+        self.pd0 = interp1d(ldelta, phi)(ld)
+
+    def _check_mdef(self, mdef):
+        if isinstance(mdef.Delta, str):
+            return True
+        elif (mdef.Delta < 200.) or (mdef.Delta > 3200.) or \
+             (mdef.rho_type != 'matter'):
+            return True
+        return False
+
+    def _get_fsigma(self, cosmo, sigM, a, lnM):
+        nu = 1.686 / sigM
+        pa = self.pa0 * a**(-0.27)
+        pb = self.pb0 * a**(-0.20)
+        pc = self.pc0 * a**0.01
+        pd = self.pd0 * a**0.08
+        return nu * self.pA0 * (1 + (pb * nu)**(-2 * pd)) * \
+            nu**(2 * pa) * np.exp(-0.5 * pc * nu**2)
+
+
+class MassFuncBocquet16(MassFunc):
+    """ Implements mass function described in arXiv:1502.07357.
+
+    Args:
+        cosmo (:obj:`Cosmology`): A Cosmology object.
+        mass_def (:obj:`MassDef`): a mass definition object.
+            this parametrization accepts SO masses with
+            Delta = 200 (matter, critical) and 500 (critical).
+            If `None`, Delta = 200 (matter) will be used.
+    """
+    name = 'Bocquet16'
+
+    def __init__(self, cosmo, mass_def=None, hydro=True):
```

Comment:
Is with hydro the right default here?

Reply:
Also no doc string for hydro

Reply:
I think `hydro = True` is the more accurate version (this is also the default in Colossus).
Docs added.